packages/@sanity/types/src/schema/types.ts

```diff
@@ -84,7 +84,7 @@ export interface BaseSchemaType {
   description?: string
   type?: SchemaType
   liveEdit?: boolean
-  readOnly?: ConditionalProperty
+  readOnly?: ConditionalProperty | undefined
```

Comment:
This is typed to be optional, does it need the extra `undefined` at the end?

Reply:
true